### PR TITLE
Add VerifySemantic to bindings

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Add VerifySemantic to bindings.

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Running js and python examples calling it
Python code:

```py
from dacite import from_dict
from iota_sdk import Utils, InputSigningData, TransactionPayload

prepared = {"essence": {"type": 1, "networkId": "7784367046153662236", "inputsCommitment": "0x7a40ce11802ca92397ac29df1fdf82047a806ddd5bbf25c04a960189b8035b35", "inputs": [{"type": 0, "transactionId": "0xba28a28e109a096a8b6347a365aaee1cf3988cbd52e58db3000b58ea8847a528", "transactionOutputIndex": 0}], "outputs": [{"type": 3, "amount": "1000000", "unlockConditions": [{"type": 0, "address": {"type": 0, "pubKeyHash": "0xdeb2e6f8e78055259a27b889a3b9a57dfb321cbe73b5c54aea1542b440400524"}}]}, {"type": 3, "amount": "999000000", "unlockConditions": [{"type": 0, "address": {"type": 0, "pubKeyHash": "0xdeb2e6f8e78055259a27b889a3b9a57dfb321cbe73b5c54aea1542b440400524"}}]}]}, "inputsData": [{"output": {"type": 3, "amount": "1000000000", "unlockConditions": [{"type": 0, "address": {
    "type": 0, "pubKeyHash": "0xdeb2e6f8e78055259a27b889a3b9a57dfb321cbe73b5c54aea1542b440400524"}}]}, "outputMetadata": {"blockId": "0x24be1fa26abdb1c50f41856bbdf68a18e7982c3dd263196e2ce72e0cf27a4d91", "transactionId": "0xba28a28e109a096a8b6347a365aaee1cf3988cbd52e58db3000b58ea8847a528", "outputIndex": 0, "isSpent": False, "milestoneIndexBooked": 400894, "milestoneTimestampBooked": 1699639723, "ledgerIndex": 400942}, "chain": {"coinType": 4219, "account": 0, "change": 0, "addressIndex": 0}}], "remainder": {"output": {"type": 3, "amount": "999000000", "unlockConditions": [{"type": 0, "address": {"type": 0, "pubKeyHash": "0xdeb2e6f8e78055259a27b889a3b9a57dfb321cbe73b5c54aea1542b440400524"}}]}, "chain": {"coinType": 4219, "account": 0, "change": 0, "addressIndex": 0}, "address": {"type": 0, "pubKeyHash": "0xdeb2e6f8e78055259a27b889a3b9a57dfb321cbe73b5c54aea1542b440400524"}}}
signed = {"type": 6, "essence": {"type": 1, "networkId": "7784367046153662236", "inputsCommitment": "0x7a40ce11802ca92397ac29df1fdf82047a806ddd5bbf25c04a960189b8035b35", "inputs": [{"type": 0, "transactionId": "0xba28a28e109a096a8b6347a365aaee1cf3988cbd52e58db3000b58ea8847a528", "transactionOutputIndex": 0}], "outputs": [{"type": 3, "amount": "1000000", "unlockConditions": [{"type": 0, "address": {"type": 0, "pubKeyHash": "0xdeb2e6f8e78055259a27b889a3b9a57dfb321cbe73b5c54aea1542b440400524"}}]}, {
    "type": 3, "amount": "999000000", "unlockConditions": [{"type": 0, "address": {"type": 0, "pubKeyHash": "0xdeb2e6f8e78055259a27b889a3b9a57dfb321cbe73b5c54aea1542b440400524"}}]}]}, "unlocks": [{"type": 0, "signature": {"type": 0, "publicKey": "0xb676bc347e3d02810b5f7641841ede6281680b36c3f92b956372a193caeee7f8", "signature": "0x65a1baff779371605f209378f14d70c8730b8394c124845d88818512105c3da3ace504396ef8c7dc80e25abced8e05f041bec3765c236e87f805e2a8938dfe09"}}]}


conflict = Utils.verify_semantic([from_dict(InputSigningData,
                                            i) for i in prepared['inputsData']], from_dict(TransactionPayload, signed), 1699532757)
print(conflict)

```

### CODE REVIEW

File: bindings/core/src/method/utils.rs [hunk 1]
```diff
@@ -159,4 +162,10 @@ pub enum UtilsMethod {
     /// Returns the hex representation of the serialized output bytes.
     #[serde(rename_all = "camelCase")]
     OutputHexBytes { output: OutputDto },
+    /// Verifies the semantic of a transaction.
+    VerifySemantic {
```
Comment (thibault-martinez) [inline bindings/core/src/method/utils.rs]: Maybe we should call it VerifyTransactionSemantic 🤔

Comment (Thoralf-M) [reply bindings/core/src/method/utils.rs]: Fine for me, but we can only rename in the bindings

File: bindings/python/iota_sdk/utils.py [hunk 2]
```diff
@@ -218,6 +219,17 @@ def verify_secp256k1_ecdsa_signature(
             'message': message,
         })
 
+    @staticmethod
+    def verify_transaction_semantic(
+            inputs: List[InputSigningData], transaction: TransactionPayload, time: int) -> bool:
```
Comment (thibault-martinez) [inline bindings/python/iota_sdk/utils.py]: bool?

Comment (Thoralf-M) [reply bindings/python/iota_sdk/utils.py]: No

**Review by thibault-martinez — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

**Pull request merged**